src/lib/Parser.hs

```diff
@@ -336,20 +336,23 @@ decl = do
 interfaceInstance :: Parser UDecl
 interfaceInstance = do
   keyWord InstanceKW
-  (lhs, pos) <- withPos simpleLet
+  (p, pos) <- withPos $ letPat <|> parens pat
+  ann <- annot uType
```

Comment:
Didn't you miss the colon before the type?

Reply:
Nope, "annot" actually parses a colon!